File: 881-boats-to-save-people/881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int i=0, j=people.length-1, c=0;
        
        while(i<=j) {
            if((people[i]+people[j]) <= limit) {
                c++;
                i++;
                j--;
            }
            else {
                c++;
                j--;
            }
        }
        return c;
    }
}

File: 881-boats-to-save-people/881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        i, j = 0, len(people)-1
        c = 0
        while i<=j:
            if(people[i]+people[j] <= limit):
                c += 1
                i += 1
                j -= 1
                
            else:
                c += 1
                j -= 1
        return c